src/focus.service.ts

```diff
@@ -421,10 +380,17 @@ export class FocusService {
 
   public createArcEvent(direction: Direction): ArcEvent {
     const directional = isDirectional(direction);
+    let element: HTMLElement;
+    if (directional) {
+      element = this.findNextFocusByRaycast(direction);
+      if (!element) {
+        element = this.findNextFocus(direction);
```

Comment:
The upper bound on this is about 9ms, since we run at most 30 checks in the raycast. In the benchmark, the ray check improved benchmark runtime by about 35% (~1.5x faster)